File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/activity/base/GGVModelActivity.java
package io.ganguo.demo.view.activity.base;

import android.databinding.ViewDataBinding;

import io.ganguo.viewmodel.common.HeaderItemViewModel;
import io.ganguo.viewmodel.common.HeaderViewModel;
import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.vmodel.ViewModelActivity;

/**
 * <p>
 * 基类- 用于封装一些通用的函数
 * </p>
 * Created by leo on 2018/7/30.
 */
public abstract class GGVModelActivity<T extends ViewDataBinding, B extends BaseViewModel> extends ViewModelActivity<T, B> {


    /**
     * function：create header ViewModel
     *
     * @return
     */
    protected HeaderViewModel onCreateHeaderVModel() {
        return new HeaderViewModel.Builder()
                .appendItemCenter(new HeaderItemViewModel
                        .TitleItemViewModel(getClass().getSimpleName()))
                .appendItemLeft(new HeaderItemViewModel
                        .BackItemViewModel(this))
                .build();
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/activity/base/GGVModelActivity.kt
package io.ganguo.demo.view.activity.base

import androidx.databinding.ViewDataBinding
import io.ganguo.demo.R
import io.ganguo.viewmodel.pack.base.activity.BaseVModelActivity

import io.ganguo.viewmodel.pack.common.HeaderItemViewModel
import io.ganguo.viewmodel.pack.common.HeaderViewModel
import io.ganguo.viewmodel.core.BaseViewModel

/**
 *
 *
 * 基类- 用于封装一些通用的函数
 *
 * Created by leo on 2018/7/30.
 */
abstract class GGVModelActivity<T : ViewDataBinding, B : BaseViewModel<*>> : BaseVModelActivity<T, B>() {


    /**
     * function：create header ViewModel
     *
     * @return
     */
   open fun newHeaderVModel(): HeaderViewModel {
        return HeaderViewModel.Builder()
                .appendItemCenter(HeaderItemViewModel.TitleItemViewModel(javaClass.simpleName))
                .appendItemLeft(HeaderItemViewModel
                        .BackItemViewModel(this)
                        .paddingRes(R.dimen.dp_19)
                        .drawableRes(getBackDrawableRes()))
                .build()
    }

    /**
     * 返回按钮资源
     * @return Int
     */
    protected open fun getBackDrawableRes(): Int {
        return R.drawable.ic_back
    }
}
